apps/console/src/index.tsx

```diff
@@ -36,6 +37,37 @@ import Theme from "./theme";
 // Set the runtime config in the context.
 ContextUtils.setRuntimeConfig(Config.getDeploymentConfig());
 
+/**
+ * TODO: Use Monaco with the webpack plugin.
+ * {@link https://github.com/wso2-enterprise/asgardeo-product/issues/23937}
+ *
+ * Function to check the status of the Monaco CDN.
+ * If the CDN is not available, the default CDN will be used.
+ */
+const checkCDNStatus = async () => {
+    try {
+        const response: Response = await fetch("https://cdn.jsdelivr.net/npm/monaco-editor@0.36.1/min/vs/loader.js");
+
+        if (response.ok) {
+            loader.config({
+                paths: {
+                    vs: "https://cdn.jsdelivr.net/npm/monaco-editor@0.36.1/min/vs"
+                }
+            });
+        } else {
+            loader.config({
+                paths: {
+                    vs: "https://cdnjs.cloudflare.com/ajax/libs/monaco-editor/0.36.1/min/vs"
+                }
+            });
+        }
+    } catch (error) {
+        // Use default CDN.
```

Comment:
When the internet is not reachable, we can configure to load a locally available loader script to be loaded in the catch block.

To do this, we first need to add "loader.js" to console/public folder.

```suggestion
        // Use default CDN.        
        const response: Response = await fetch(contextPathGlobal + "loader.js");

        if (response.ok) {
            loader.config({
                paths: {
                    vs: contextPathGlobal + "loader.js"
                }
            })
        }
```

@brionmario what are your thoughts on this?

Reply:
loader just a script used to download Monaco resources, we will still need internet for this AFIK...

Reply:
If the script is packed with console bundle, the app will not need to access internet to get the script.

Reply:
@pavinduLakshan So your suggestion is to maintain a `minified` monaco bundle as a static file? I hope `loader.js` is just a generic term used for the actual monaco minified bundle?

Is there a one single such minified bundle available to download?

Reply:
> Is there a one single such minified bundle available to download?

We can download it from https://cdn.jsdelivr.net/npm/monaco-editor@0.36.1/min/vs or extract from the monaco-editor npm package, can't we?

Reply:
https://cdn.jsdelivr.net/npm/monaco-editor@0.36.1/min/vs/loader.js this is the path the loader which is a script used to download the Monaco resources. We will still need internet access to get the Monaco resources AFIK. Could you clarify this suggestion a bit more? @pavinduLakshan

Reply:
If the loader.js script is just for downloading stuff from internet, we wouldn't be able to follow my suggested approach. Sorry for the noise.